man/systemd.unit.xml

```diff
@@ -1031,6 +1033,15 @@
         whether a certain path exists, is a regular file and marked
         executable.</para>
 
+        <para><varname>ConditionUID=</varname> takes a numeric
+        <literal>UID</literal>. This condition may be used to check
+        wether the service manager is running as the given user id.
+        This option is only meaningful for the user managers.</para>
+
+        <para><varname>ConditionGID=</varname> is similar
+        to <varname>ConditionUID=</varname> but verifies that the
+        service manager is in the specified group.</para>
```

Comment:
Well, ConditionGroup will check against a list of groups anyway, as each process has an auxiliary group list. Hence in in_gid() it appeared natural to also check both gid and egid, following the logic that if we test against multiple possible anyway, we better should do that comprehensively. Or to say this differently: the effect of egid is "bigger" than for entries of the auxiliary group list, and hence it would be surprising if we check the latter but not egid too.

And yeah, following that logic we should probably check both uid/euid for the ConditionUser= case, otherwise things would be weirdly asymmetric, you are right about that. Of course, this should be documented.

IRL I figure this doesn't all matter much, as systemd instances are unlikely to run in a context where gid != egid or uid != euid, but i figure it doesn't hurt to be systematic here...